### PR TITLE
Validate version 5 of clspv reflection

### PR DESCRIPTION
**Pull request by alan-baker**
* Add validation for instructions in versions 2 through 5
* Change the instruction reported for remaining indices on an access chain to the access chain itself for clarity

### CODE REVIEW

File: source/val/validate_extensions.cpp [hunk 1]
```diff
@@ -273,12 +287,14 @@ spv_result_t ValidateOperandDebugType(
 }
 
 spv_result_t ValidateClspvReflectionKernel(ValidationState_t& _,
-                                           const Instruction* inst) {
+                                           const Instruction* inst,
+                                           uint32_t version) {
+  const auto inst_name = ReflectionInstructionName(_, inst);
```
Comment (dneto0) [inline source/val/validate_extensions.cpp:292]: nitpick.  I'd defer constructing the string until the time I know it's needed.
You can do that by making this a lambda, and then only calling it when generating the diagnostic.

File: source/val/validate_extensions.cpp [hunk 2]
```diff
@@ -691,11 +974,31 @@ spv_result_t ValidateClspvReflectionInstruction(ValidationState_t& _,
...
+    case NonSemanticClspvReflectionProgramScopeVariablePointerRelocation:
+      return ValidateClspvReflectionPointerRelocation(_, inst);
+    case NonSemanticClspvReflectionImageArgumentInfoChannelOrderPushConstant:
+    case NonSemanticClspvReflectionImageArgumentInfoChannelDataTypePushConstant:
+      return ValidateClspvReflectionImageMetadataPushConstant(_, inst);
+    case NonSemanticClspvReflectionImageArgumentInfoChannelOrderUniform:
+    case NonSemanticClspvReflectionImageArgumentInfoChannelDataTypeUniform:
+      return ValidateClspvReflectionImageMetadataUniform(_, inst);
+    case NonSemanticClspvReflectionConstantDataPointerPushConstant:
+    case NonSemanticClspvReflectionProgramScopeVariablePointerPushConstant:
```
Comment (dneto0) [inline source/val/validate_extensions.cpp:994]: There is a typo in the spec text. It's spelled



ProgramScopeVariable**s**PointerPushConstant

**Review by dneto0 — APPROVED** [review]
Thanks!

File: source/val/validate_extensions.cpp (hunk 2 above)
Comment (alan-baker) [reply source/val/validate_extensions.cpp:994]: I made that mistake several times writing this code. I'll update the spec to match the headers.

Comment (alan-baker) [reply source/val/validate_extensions.cpp:994]: See https://github.com/KhronosGroup/SPIRV-Registry/pull/183

**Pull request merged**